fireant/queries/builder.py

```diff
@@ -297,16 +301,47 @@ def sql(self):
         Serializes this query builder as a set of SQL queries.  This method will always return a list of one query since
         only one query is required to retrieve dimension choices.
 
-        This function only handles dimensions (select+group by) and filtering (where/having), which is everything needed
-        for the query to fetch choices for dimensions.
-
         The slicer query extends this with metrics, references, and totals.
         """
+
+        def in_hint_table(dim):
+            return all(dim_field.name in hint_column_names for dim_field in dim.definition.fields())
+
+        hint_column_definitions = self.dataset.database.get_column_definitions(
+              self.dataset.hint_table._schema._name,
+              self.dataset.hint_table._table_name
+        ) if self.dataset.hint_table else []
+
+        hint_column_names = [hint_column_definition[0] for hint_column_definition in hint_column_definitions]
+
+        dimension = self._dimensions[0]
```

Comment:
Out of interest, why are we just getting the first dimension here?

Reply:
Only the first dimension is the actual dimension we are fetching choices for, the remaining are display dimensions for it. However, I just pushed a change that also includes the display dimensions in the hint dimension resolution, so that the hint table will only be used if the display dimensions are matched as well.